lib/hawkular/hawkular_client.rb

```diff
@@ -18,6 +16,10 @@ def initialize(hash)
       @state = hash
     end
 
+    def respond_to_missing?(method_name, include_private = false)
+      method_name.to_s.start_with?('inventory_', 'alerts_', 'operations_', 'tokens_', 'prometheus_') || super
```

Comment:
I rechecked this one, and I think that you still need to delegate to respond_to? to the relevant sub_client. Since the prefix won't guarantee that the method exists in the subclient.

Reply:
@xeviknal ^

Reply:
@israel-hdez  I got what you said. Then answer to `respond_to?` becomes more precise when checking to the corresponding subclient.

In the proposed solution, i used `respond_to?` inside `method_missing`. I am wondering if this is a bad approach. Does anybody know?